---
title: "Providence Therapeutics Draft Report"
author: "GSI Bioinformatic Support"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
params:
  ext: ""
  sample: ""
  library: ""
  flowcell: ""
  ref_bases: ""
  error: ""
  consensus: ""
  reference: ""
  construct: ""
  
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.pos= "H")
```

\fontfamily{lmr}\selectfont 

OICR Sample: **`r params$sample`** 
\newline
OICR Library: **`r params$library`**
\newline
External Name: **`r params$ext`**
\newline
Run: **`r params$flowcell`**


## \textcolor{blue}{Reference sequence}

  ``` {r seq, echo=FALSE, warning=FALSE}
library(knitr)
library(kableExtra)
 summary_table<-read.table("ref.txt",header=TRUE,sep="\t")
 kable(summary_table, col.names = colnames(summary_table), row.names = FALSE,  "latex", booktabs = T)
 ```
Total Bases:`r params$ref_bases`
\newline

 \fontfamily{lmtt}\selectfont

**`r params$ext`** reference provided by Providence Therapeutics
```{r fastq_seq, echo=FALSE, warning=FALSE, comment=NA, fig.align='left' }
library(knitr)


formatDNA<-function(fasta.file){
  dna<-scan(fasta.file,what="character")
  dna<-dna[-1]
  dna<-paste(dna,collapse="")

  dna.lines<-strsplit(dna, '(?<=.{60})', perl = TRUE)[[1]]
  ### set the positions
  n<-60
  names(dna.lines)<-seq(from=1,by=n,length.out=length(dna.lines))
  dna.df<-as.data.frame(dna.lines)

  dna.chunked<-apply(dna.df,1,function(x){paste(strsplit(x, "(?<=.{10})", perl = TRUE)[[1]],collapse=" ")})
  dna.chunked.df<-as.data.frame(dna.chunked)
  colnames(dna.chunked.df)<-NULL
  dna.chunked.df
}

fasta.file<-"ref.fa"
dna<-formatDNA(fasta.file)
print(dna,right=FALSE)
```

\leavevmode\newline 

\fontfamily{lmr}\selectfont 

## \textcolor{blue}{Summary}
Table below shows the run summary information for the analyzed sample:
\newline 

``` {r adapter_summary, echo=FALSE, warning=FALSE}
library(knitr)
library(kableExtra)

 summary_table<-read.table("adapter_trimming.txt",header=TRUE,sep="\t")
 kable(summary_table, col.names = colnames(summary_table), row.names = FALSE,  "latex", booktabs = T)%>%
kable_styling(latex_options = c("striped", "scale_down"))
```
\leavevmode\newline

``` {r summary, echo=FALSE, warning=FALSE}
library(knitr)
library(kableExtra)

 summary_table<-read.table("summary.txt",header=TRUE,sep="\t")
 kable(summary_table, col.names = colnames(summary_table), row.names = FALSE,  "latex", booktabs = T)%>%
kable_styling(latex_options = c("striped", "scale_down"))
```
##  \textcolor{blue}{Read Length Distribution}

```{r read_dist, echo=FALSE, out.width = '65%'}
knitr::include_graphics("read_dist.png")
```

##  \textcolor{blue}{Insert Size Distribution}
```{r insert_size_dist, echo=FALSE, out.width = '65%'}
knitr::include_graphics("insert_size_dist.png")
```

##  \textcolor{blue}{High-quality Bases [>Q13]}

```{r genome_fracture_coverage, echo=FALSE, out.width = '65%'}
knitr::include_graphics("coverage.png")
```

##  \textcolor{blue}{Error Distribution}

```{r error_dist, echo=FALSE, out.width = '65%'}
knitr::include_graphics("error_dist.png")
```
\leavevmode\newline 
The average error rate was **`r params$error`** 

\leavevmode\newline 

## \textcolor{blue}{Mutations}

The presence of mutations was investigated using bcftools variant calling algorithm. No variants were detected.

## \textcolor{blue}{Consensus}
The consensus sequence was generated from the aligned reads using Bcftools.This consensus sequence contains the variants determined by the reads.

The consensus sequence matches the reference sequence with **`r params$consensus`** sequencing identity.
\newline

Query: **`r params$library`**
Sbjct: **`r params$ext`**

```{r consensus, echo=FALSE, engine='bash', comment=''}
cat blast_alignment.txt
```

## \textcolor{blue}{ORF Summary}

``` {r orf_summary, echo=FALSE, warning=FALSE}
library(knitr)
library(kableExtra)

 summary_table<-read.table("orf_summary.txt",header=TRUE,sep="\t")
 kable(summary_table, col.names = colnames(summary_table), row.names = FALSE,  "latex", booktabs = T, longtable = TRUE)%>%
  kable_styling(bootstrap_options = "striped", full_width = F, position = "left")
 
```


```{r orf, echo=FALSE, engine='bash', comment=''}
cat orf.txt
```


## \textcolor{blue}{Consensus of the Construct}
The DNA construct was also sequenced and a consensus sequence was generated from the aligned reads using Bcftools. Please refer to the **`r params$construct`** report
\newline

